src/agent0/core/hyperdrive/policies/simple_lp_test.py

```diff
@@ -0,0 +1,98 @@
+    initial_pool_config = ILocalHyperdrive.Config(
+        initial_liquidity=FixedPoint("100"),
+        initial_variable_rate=FixedPoint("0.01"),
+        initial_fixed_apr=FixedPoint("0.05"),
+        position_duration=60 * 60 * 24 * 30,  # 1 month
+        checkpoint_duration=60 * 60 * 24,  # 1 day
+        curve_fee=FixedPoint("0.01"),
+        flat_fee=FixedPoint("0.0005"),
+        governance_lp_fee=FixedPoint("0.15"),
+        governance_zombie_fee=FixedPoint("0.03"),
+    )
+    interactive_hyperdrive = ILocalHyperdrive(chain, initial_pool_config)
+
+    # Deploy LP agent & add base liquidity
+    pnl_target = FixedPoint("8.0")
+    delta_liquidity = FixedPoint("100")
+    minimum_liquidity_tokens = FixedPoint("100")
+    lp_agent = interactive_hyperdrive.init_agent(
+        base=FixedPoint("1_000_000"),
+        eth=FixedPoint("1_000"),
+        name="Lisa",
+        policy=PolicyZoo.simple_lp,
+        policy_config=PolicyZoo.simple_lp.Config(
+            pnl_target=pnl_target,
+            delta_liquidity=delta_liquidity,
+            minimum_liquidity_tokens=minimum_liquidity_tokens,
+        ),
+    )
+    _ = lp_agent.add_liquidity(base=FixedPoint("10_000"))
+
+    # Do dumb trades until the LP makes a move; that move should be adding liquidity
+    hyperdrive_agent0 = interactive_hyperdrive.init_agent(
+        base=FixedPoint("1_000_000"), eth=FixedPoint("1_000"), name="Bob"
+    )
+    trade_event_list = []
+    while len(trade_event_list) == 0:
+        trade_amount = FixedPoint("1_000")
+        hyperdrive_agent0.add_funds(base=trade_amount)
+        open_event = hyperdrive_agent0.open_short(trade_amount)
+        chain.advance_time(datetime.timedelta(weeks=1), create_checkpoints=False)
+        hyperdrive_agent0.close_short(open_event.maturity_time, open_event.bond_amount)
+        trade_event_list = lp_agent.execute_policy_action()
+
+    # only one trade per action execution
+    assert len(trade_event_list) == 1
+    # always should be add liquidity
+    assert isinstance(trade_event_list[0], AddLiquidity)
+    # always should be close to delta_liquidity
+    assert isclose(trade_event_list[0].lp_amount, delta_liquidity, abs_tol=FixedPoint("1.0"))
+
+    # Do smart trades until the LP removes liquidity
+    # It's possible the LP could add liquidity in the first couple of trades,
+    # depending on how much they influence the average PNL.
+    hyperdrive_agent1 = interactive_hyperdrive.init_agent(
+        base=FixedPoint("1_000_000"), eth=FixedPoint("1_000"), name="Bob"
+    )
+    trade_event_list = []
+    removed_liquidity = False
+    while not removed_liquidity:
+        trade_amount = FixedPoint("1_000")
+        hyperdrive_agent1.add_funds(base=trade_amount)
+        interactive_hyperdrive.set_variable_rate(FixedPoint("0.8"))
+        open_event = hyperdrive_agent1.open_long(trade_amount)
+        chain.advance_time(datetime.timedelta(seconds=100), create_checkpoints=False)
```

Comment:
not sure why this and the `interactive_hyperdrive.set_variable_rate(FixedPoint("0.8"))` above are needed

we're making LPs lose money by selling fixed rates above the variable rate, right? so setting variable rate to 0 should suffice

Reply:
we set the rate high so that the open long gets that fixed rate, which means the LP has to supply it. And then we set the variable to zero before advancing a larger amount of time (1 week) so that the LP doesn't get any kickback from the variable while they're providing the fixed. It's a shit scenario for them, which was the intention so that I'd be sure they pulled their liquidity.

Reply:
correct, they're independent. I don't know why I had that mixed up! Good catch.